coalib/output/ConsoleInteraction.py

```diff
@@ -636,26 +638,85 @@ def ask_for_action_and_apply(console_printer,
                             it makes sense that the user may choose to execute
                             another action, False otherwise.
     """
+    flag = False
     action_name, section = print_actions(console_printer, section,
                                          metadata_list, failed_actions)
+
+    actions_name_dict = {'(A)': 'ApplyPatchAction', '(I)': 'IgnoreResultAction',
+                         '(O)': 'OpenEditorAction', '(T)': 'PrintAspectAction',
+                         '(D)': 'PrintDebugMessageAction',
+                         '(M)': 'PrintMoreInfoAction', '(S)': 'ShowPatchAction'}
     if action_name is None:
         return False
 
-    chosen_action = action_dict[action_name]
-    try:
-        chosen_action.apply_from_section(result,
-                                         file_dict,
-                                         file_diff_dict,
-                                         section)
-        console_printer.print(
-            format_lines(chosen_action.SUCCESS_MESSAGE, symbol='['),
-            color=SUCCESS_COLOR)
-        failed_actions.discard(action_name)
-    except Exception as exception:  # pylint: disable=broad-except
-        logging.error('Failed to execute the action {} with error: {}.'.format(
-            action_name, exception))
-        failed_actions.add(action_name)
-    return True
+    if action_name == 'ChainPatchAction':
```

Comment:
It's more maintainable if you break it down into little methods than a huge sequence like this.